Library/src/Multiplayer/SpaceEntitySystem.cpp

```diff
@@ -419,6 +409,25 @@ void SpaceEntitySystem::DestroyEntity(SpaceEntity* Entity, CallbackHandler Callb
 
 	CSP_DELETE(Keys);
 
+	csp::common::Array<ComponentUpdateInfo> Info;
+
+	RootHierarchyEntities.RemoveItem(Entity);
+
+	// Manually process the parent updates locally
+	// We want this callback to fire before the deletion so clients can react to children first
+	auto ChildrenToUpdate = Children;
+
+	for (size_t i = 0; i < ChildrenToUpdate.Size(); ++i)
+	{
+		ChildrenToUpdate[i]->ParentId = nullptr;
```

Comment:
[question] Is there a reason for not using RemoveParentEntity? (not a leading question at all, when I saw that I just thought there may be a function that does it already and I found that one)

Reply:
RemoveParentEntity is used when going through the patch process, as it also sets ShouldUpdateParent. We don't need to do that here, as we're updating manually